frontend/lib/browser-storage-base.tsx

```diff
@@ -121,3 +160,32 @@ export function createUpdateBrowserStorage<T extends BaseBrowserStorageSchema>(
     return null;
   };
 };
+
+/**
+ * Creates a `useBrowserStorage` React Hook that can be used in a way that
+ * is similar to `useState()`, only it returns/updates the value of browser
+ * storage.
+ * 
+ * Before a component is mounted, this will actually return the storage's
+ * default value to ensure that rendering is identical on server and client.
+ */
+export function createUseBrowserStorage<T extends BaseBrowserStorageSchema>(
+  browserStorage: BrowserStorage<T>
+) {
+  const useBrowserStorage = (): [T, (updates: Partial<T>) => void] => {
+    const [state, setState] = useState(browserStorage.defaultValue);
+    const updateState = useCallback((updates: Partial<T>) => {
+      browserStorage.update(updates);
+    }, []);
+
+    useEffect(() => {
+      setState(browserStorage.getAll());
+      return browserStorage.listenForChanges((newState) => {
+        setState(newState);
+      });
+    }, []);
```

Comment:
I thought you mentioned that adding this empty array as a second argument in `useEffect` will only fire the hook once, when the IssueAreaLink component mounts/unmounts? But this isn't the desired behavior as a change in the browser storage occurs after all of the IssueAreaLink's have already mounted.

Reply:
Ah, so this is a bit funky because what the `useEffect` call here is really doing is setting up an _event listener_ on browser storage, and using the lifetime of the component using the hook  to determine the lifetime of the listener.  That is, on component mount, it starts listening for changes to browser storage, and when they happen, it instantly triggers a change in the component that's using the hook; and then once the component unmounts, it stops listening to changes in browser storage.

Does that make sense?